src/software/firmware/srcs/pressure_controller.cpp

```diff
@@ -132,15 +132,20 @@ void PressureController::initRespiratoryCycle() {
 void PressureController::endRespiratoryCycle() {
     checkCycleAlarm();
 
-    if (m_blower_increment == 0u) {
-        if (m_plateauPressure > (m_maxPlateauPressureCommand * 102u / 100u)) {
-            uint16_t plateauDiff = (((m_plateauPressure - m_maxPlateauPressureCommand) * 2u) / 10u);
-            onPeakPressureDecrease(min(plateauDiff, MAX_PEAK_INCREMENT));
-        } else if (m_plateauPressure < (m_maxPlateauPressureCommand * 98u / 100u)) {
-            uint16_t plateauDiff = (((m_maxPlateauPressureCommand - m_plateauPressure) * 2u) / 10u);
-            onPeakPressureIncrease(min(plateauDiff, MAX_PEAK_INCREMENT));
-        }
-    }
+        // Case plateau is not detected or plateau is too close to Peep pressure
+    if (m_plateauPressure == 0 || abs(m_plateauPressure-m_peep) < 10 ){
+        m_plateauPressure = UINT16_MAX;
+    } 
+    // Disable peak pressure control
+    /*else if (m_plateauPressure > (m_maxPlateauPressureCommand * 102u / 100u)) {
+        uint16_t plateauDiff = (((m_plateauPressure - m_maxPlateauPressureCommand) * 2u) / 10u);
+        onPeakPressureDecrease(min(plateauDiff, MAX_PEAK_INCREMENT));
+
+    // Increase peak pressure only if blower is stable.
+    } else if (m_plateauPressure < (m_maxPlateauPressureCommand * 98u / 100u) && m_blower_increment == 0u) {
+        uint16_t plateauDiff = (((m_maxPlateauPressureCommand - m_plateauPressure) * 2u) / 10u);
+        onPeakPressureIncrease(min(plateauDiff, MAX_PEAK_INCREMENT));
+    }*/
```

Comment:
If the revert is needed, we don't need to keep the removed code in comments. Git is here and save us from loosing interesting code.

Reply:
And also done because it told myself the same!